### ISSUE TITLE
UVCal method to inherit metadata from a UVData object

### ISSUE THREAD

**Issue by bhazelton** [RadioAstronomySoftwareGroup/pyuvdata#753]
Be able to copy all relevant metadata from a UVData object into a UVCal object.

**Comment by jsdillon** [comment]
+1 

This would be useful for hera_cal, since we have a few instances where were make UVCal objects from scratch and end up passing a dozen or so parameters in, mostly from some data object. I think this would be a cleaner and less bug-prone approach. Not a need to have, but certainly a nice to have.

**Comment by bhazelton** [comment]
I'm starting to think about doing this, trying to collect the list of things that should be set:
- telescope_name
- telescope_location (I'll add this as an optional parameter to address #735)
- antenna_number
- antenna_name
- Nants_telescope
- x_orientation

I'm less sure about the frequency and time arrays, maybe setting them should be optional? If you're only solving for a single calibration solution across a range of times you only want one time, a longer `integration_time` and you probably want to set the `time_range`.

I'm also unsure about whether the extra_keywords should be copied over.

**Comment by jsdillon** [comment]
All that seems reasonable, but ideally I'd like something that uses the data file to infer the antennas, times, frequencies in data and use that to build placeholder gain_array, flag_array, etc. of the correct size and datatype.

I recognize that that's not easy to fully generalize when we're dealing with the full range of what a UVData object could be, but maybe there's some relatively small number of things that would have to be specified to make that happen? Times comes to mind as a big issue due to BDA. And of course, I'd be fine with this functionality being optional and default off.

Basically, I'd like this to create a UVCal object that could then be applied to the UVData object without an error. If that's not in the cards (since it's substantially more work), then this issue is a lot lower priority to me (and it wasn't super high to begin with).

**Comment by bhazelton** [comment]
That's not too hard, I think it can be done. We can get something in that has the options you need now and add more if BDA needs different things.

**Comment by bhazelton** [comment]
maybe it would help to also add a metadata_only property in UVCal like the one in UVData.

**Comment by bhazelton** [comment]
Let's use keywords to allow the time and frequency parameters to be set, if they aren't set just copy from the UVData object. Allow for metadata_only initialization. Possibly allow a list of attributes to exclude copying over.

**Comment by bhazelton** [comment]
Should this be a method on UVCal (named something like `initialize_from_uvdata`) or should it be a utility function (named something like `uvcal_from_uvdata`)?

**Comment by jsdillon** [comment]
I think both options are reasonable. No strong preference.
